File: src/cviz/control/tcp/TCPControlInterface.java
package cviz.control.tcp;

import cviz.TimelineManager;
import cviz.TimelineState;
import cviz.config.Config;
import cviz.control.IControlInterface;
import cviz.state.State;

import java.util.concurrent.ConcurrentLinkedQueue;

public class TCPControlInterface implements IControlInterface {

    private final ControlServer server;
    private final TimelineManager manager;

    public TCPControlInterface(Config config, TimelineManager manager){
        this.manager = manager;

        this.server = new ControlServer(config, manager);
        new Thread(this.server).start();
    }

    @Override
    public void notifyState(State state) {
        // Nothing to do. state is broadcasted at regular interval without needing a notify
    }

    @Override
    public void run() {
        while(true){
            State[] state = manager.getCompleteState();
            for (State st : state) {
                server.sendState(st);
            }

            try {
                Thread.sleep(10);
            } catch (InterruptedException e) {
            }
        }
    }
}

File: CViz/Control/Tcp/TcpControlInterface.cs
using System;
using System.Threading;
using CViz.Util;

namespace CViz.Control.Tcp
{
    class TcpControlInterface : IControlInterface
    {
        private readonly TimelineManager _manager;
        private readonly ControlServer _server;

        public TcpControlInterface(Config.Config config, TimelineManager manager)
        {
            _manager = manager;

            _server = new ControlServer(config, manager);
            new Thread(_server.Run).Start();
        }

        public void NotifyState(State.State state)
        {
            // Nothing to do. state is broadcasted at regular interval without needing a notify
        }

        public void Run()
        {
            while (true)
            {
                _manager.GetCompleteState().ForEach(_server.SendState);

                try
                {
                    Thread.Sleep(10);
                }
                catch (Exception)
                {
                }
            }
        }
    }
}